src/cdk/table/table-layout-strategy.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Inject, InjectionToken} from '@angular/core';
+import {CdkTable} from '@angular/cdk/table/table';
+import {DOCUMENT} from '@angular/common';
+
+/** Interface for a service that constructs the DOM structure for a {@link CdkTable}. */
+export interface _TableLayoutStrategy {
+  /** Constructs the DOM structure for a native table. */
+  getNativeLayout(table: CdkTable<any>): DocumentFragment;
+  /** Constructs the DOM structure for a flex table. */
+  getFlexLayout(table: CdkTable<any>): DocumentFragment;
+}
+
+/** Injection token for {@link _TableLayoutStrategy}. */
+export const _TABLE_LAYOUT_STRATEGY =
+    new InjectionToken<_TableLayoutStrategy>('_TableLayoutStrategy');
+
+
+export class _DefaultTableLayoutStrategy implements _TableLayoutStrategy {
```

Comment:
I prefer to name classes for what they do, rather than how they're used (e.g. "Default" here). How about something like `_StandardTableLayoutStrategy`?